src/main/scala/cromwell/engine/package.scala

```diff
@@ -49,13 +49,48 @@ package object engine {
    * created
    */
   case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    // TODO: Extract this from here (there is no need to reload the configuration for each workflow)
+    // Not private because overridden in tests
+    lazy private [engine] val conf = ConfigFactory.load
+
     val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
       case Success(options: JsObject) => WorkflowOptions.fromJsonObject(options).get // .get here to purposefully throw the exception
       case Success(other) => throw new Throwable(s"Expecting workflow options to be a JSON object, got $other")
       case Failure(ex) => throw ex
     }
 
-    val backend = Backend.from(workflowOptions.getOrElse("default_backend", ConfigFactory.load.getConfig("backend").getString("backend")))
+    val props = sys.props
+    val workflowLogger = props.get("LOG_MODE") match {
+      case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
+        Paths.get(props.getOrElse("LOG_ROOT", ".")),
+        s"workflow.$id.log",
+        Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
+      )
+      case _ => NOPLogger.NOP_LOGGER
+    }
+
+    // Call Caching
+    // TODO: Add to lenthall
+    def getConfigOption(key: String): Option[Config] = if (conf.hasPath(key)) Option(conf.getConfig(key)) else None
+    private lazy val configCallCaching = getConfigOption("call-caching") map { _.getBooleanOr("enabled", DefaultCallCachingValue) } getOrElse DefaultCallCachingValue
+    private lazy val optionCacheWriting = workflowOptions.getBoolean("write-to-cache") getOrElse configCallCaching
+    private lazy val optionCacheReading = workflowOptions.getBoolean("read-from-cache") getOrElse configCallCaching
+
+    if(!configCallCaching && optionCacheWriting) {
+      workflowLogger.warn(
+        """Write to cache is enabled in the workflow options but Call Caching is disabled in this Cromwell instance.
+          |As a result the call executions from this workflow will NOT be cached.""".stripMargin)
+    }
+    if(!configCallCaching && optionCacheReading) {
+      workflowLogger.warn(
+        """Read from cache is enabled in the workflow options but Call Caching is disabled in this Cromwell instance.
```

Comment:
What I'd personally do is move the template function to the companion object and then have it take some flag (e.g. an ADT of case objects) to fill in the appropriate blanks. The only two things which change are the `Write to cache`/`Read from cache` and `will NOT` and `WILL` and they track together.

So something like this in the companion object (note, hand typing this & not thinking about names much either):

    sealed trait CacheDisabled
    case object WriteCacheDisabled extends CacheDisabled
    case object ReadCacheDisabled extends CacheDisabled

    val baseString = "is enabled in the workflow options but Call Caching is disabled in this Cromwell instance. As a result the call executions for this workflow"

    def disabledMessage(cacheDisabled: CacheDisabled) = cacheDisabled match {
      case WriteCache => s"Write to cache $baseString will NOT be cached"
      case ReadCache = s"Read from cache $baseString WILL be executed"
   }

And then in your actual code here something like:

    if (!configCallCaching) {
      if (optionCacheWriting) disabledMessage(WriteCacheDisabled)
      else if (optionCacheReading) disabledMessage(ReadCacheDisabled)
   }

I just noticed the whole workflowLogger.warn bit and didn't bother including it here, but you get the idea. Perhaps make it `logDisabledMessage` and include the log in there as well although then it wouldn't cleanly fit in the companion object.

My point is that I don't really care what the actual message is when I'm quickly scanning the code and having all of those strings there requires me to kinda parse what's going on whereas something like `logDisabeledMessage(WriteCacheDisabled)` IMO conveys the same basic information for a lot less visual noise. And it's trivial to drill into those if I care what the actual string is.

That said, I don't care a ton about this